### PR TITLE
Auth!

### LINKED ISSUE #1074: [FR] Authentication!
Hi there!

So i was trying to add Frigate to traefik - and succeeded - but then noticed; Frigate has no AUTH at all!
please add some sort of authentication if possible, or am I missing something?

### PR DESCRIPTION
**Pull request by blakeblackshear**
Todos:

- [x] bypass for exports/audio and trusted networks
- [ ] ~auth settings in home assistant integration~
- [x] update docs
- [ ] ~consider splitting salt between db and config file~
- [x] add profile endpoint to api to return user
  - [x] look for ways to make things compatible with upstream proxies (traefik-forward-auth, authelia, oauth2_proxy, home assistant addon). add logout url/endpoint for upstream auth
- [x] implement logout
- [x] fix bug in db call on refresh jwt
- [x] login screen not bringing up keyboard when installed as app in ios

### CODE REVIEW

File: web/src/LoginPage.tsx
Comment (NickM-27) [inline web/src/LoginPage.tsx:1]: can we move this to /src/pages ?

File: web/src/components/AuthForm.tsx
Comment (NickM-27) [inline web/src/components/AuthForm.tsx:1]: Can we move this to /compnents/auth folder?

File: frigate/config.py [hunk 1]
```diff
@@ -116,6 +116,37 @@ class UIConfig(FrigateBaseModel):
     )
 
 
+class UserConfig(FrigateBaseModel):
```
Comment (NickM-27) [inline frigate/config.py:119]: Seems like this may not be used anymore?

File: web/src/components/menu/AccountSettings.tsx
Comment (NickM-27) [inline web/src/components/menu/AccountSettings.tsx:1]: Maybe we should hide this icon if auth is not enabled?

File: docs/docs/configuration/authentication.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+| Mode     | Description                                                                                                                                                                     |
+| -------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `native` | (default) Use this mode if you don't implement authentication with a proxy in front of Frigate.                                                                                 |
+| `proxy`  | Use this mode if you have an existing proxy for authentication. Supports passing authenticated user downstream to Frigate for role-based authorization (future implementation). |
+
+### Native mode
+
+Frigate stores user information in it's database. Password hashes are generated using industry standard PBKDF2-SHA256 with 600,000 iterations. Upon successful login, a JWT token is issued with an expiration date and set as a cookie. The cookie is refreshed as needed automatically. This JWT token can also be passed in the Authorization header as a bearer token.
```
Comment (NickM-27) [inline docs/docs/configuration/authentication.md:19]:
```suggestion
Frigate stores user information in its database. Password hashes are generated using industry standard PBKDF2-SHA256 with 600,000 iterations. Upon successful login, a JWT token is issued with an expiration date and set as a cookie. The cookie is refreshed as needed automatically. This JWT token can also be passed in the Authorization header as a bearer token.
```

File: docs/docs/configuration/authentication.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+Users are be managed under at Settings > Authentication.
```
Comment (NickM-27) [inline docs/docs/configuration/authentication.md:21]:
```suggestion
Users are in the UI managed at Settings > Authentication.
```

**Review by NickM-27 — CHANGES_REQUESTED** [review]

**Review by NickM-27 — APPROVED** [review]

**Comment by wangeris** [comment]
thank you so much for this

**Pull request merged**